src/dsl/validation.rs

```diff
@@ -27,7 +30,42 @@ pub fn validate(
         }
     };
 
-    validation_result
+    let mut schema_validation_errors = match validation_result {
+        Ok(_) => vec![],
+        Err(errors) => errors,
+    };
+
+    let (_, expression_errors): (Vec<_>, Vec<_>) = json_check
+        .get("expectations")
```

Comment:
We would need to repeat the same thing with the `values` entry, as we do the same rhai evaluation there